### PR TITLE
PySide6 Darwin Compatibility

### PR DESCRIPTION
**Pull request by CyborgPotato**
[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [ ] There's an _[automated test](tests/default.nix)_ for this change
- [X] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [X] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

These changes make the wheels used to create pyside6 through poetry2nix able to run, the current derivation fails because of Linux specific packages such as speechd, and autoPatchelfcommands. I've tested with the majority of widgets and they appear to work on my machine.

Another issue that was run into is that the wheels that are used by default (regardless of preferWheel) cause an ordering conflict in the PythonPath, hence why pyside6 is overridden with a combined symlink tree of all the conflicting packages so that they all reside with the same priority in the same site-packages directory

### CODE REVIEW

File: overrides/default.nix [hunk 1]
```diff
@@ -2615,9 +2631,9 @@ lib.composeManyExtensions [
           pkgs.pulseaudio
           self.shiboken6
         ];
-      });
+      } else old);
```
Comment (cpcloud) [inline overrides/default.nix:2634]:
```suggestion
      });
```

File: overrides/default.nix [hunk 2]
```diff
@@ -2584,7 +2584,23 @@ lib.composeManyExtensions [
...
+              name = "pyside6-combined-addons-essentials";
+              paths = [
+                super.pyside6
+                self.pyside6-essentials
+                self.pyside6-addons
+                self.shiboken6
+              ];
+            }) else super.pyside6;
+      
+      pyside6-essentials = super.pyside6-essentials.overridePythonAttrs (old: if stdenv.isLinux then {
```
Comment (cpcloud) [inline overrides/default.nix:2603]:
```suggestion
      pyside6-essentials = super.pyside6-essentials.overridePythonAttrs (old: lib.optionalAttrs stdenv.isLinux {
```

File: overrides/default.nix [hunk 3]
```diff
@@ -2615,9 +2631,9 @@ lib.composeManyExtensions [
           pkgs.pulseaudio
           self.shiboken6
         ];
-      });
+      } else old);
 
-      pyside6-addons = super.pyside6-addons.overridePythonAttrs (old: {
+      pyside6-addons = super.pyside6-addons.overridePythonAttrs (old: if stdenv.isLinux then {
```
Comment (cpcloud) [inline overrides/default.nix:2636]:
```suggestion
      pyside6-addons = super.pyside6-addons.overridePythonAttrs (old: lib.optionalAttrs stdenv.isLinux {
```

File: overrides/default.nix [hunk 4]
```diff
@@ -2638,7 +2654,7 @@ lib.composeManyExtensions [
         postInstall = ''
           rm -r $out/${self.python.sitePackages}/PySide6/__pycache__
         '';
-      });
+      } else old);
```
Comment (cpcloud) [inline overrides/default.nix:2657]:
```suggestion
      });
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
Thanks for the PR!

If there aren't already tests for these can you add them?

**Review by CyborgPotato — COMMENTED** [review]
Thank you for looking it over! I've moved the pyside6 to out of the Darwin exclusive set, and applied your suggested changes

**Comment by cpcloud** [comment]
@CyborgPotato It looks like the macos build is failing.

**Comment by CyborgPotato** [comment]
The error from the failing [job](https://github.com/nix-community/poetry2nix/actions/runs/7490587868/job/20389949722?pr=1494) says:

> error: Missing suitable source/wheel file entry for pyside6

Which I am having difficulty reproducing on my machine (though I also don't have an x86_64 mac to test on). Any advice for this error?

**Comment by CyborgPotato** [comment]
@cpcloud I have found why the macos build aggregate failed, the pyside6 pyproject file expected a version of pyside that requires macos >= 11, while nixpkgs defaults to 10.12, making it unable to fetch the appropriate wheel.

**Comment by cpcloud** [comment]
Closing in favor of #1621.

**Pull request closed without merging**